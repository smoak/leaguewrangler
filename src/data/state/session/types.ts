export interface Session {
  readonly userToken: string;
}
