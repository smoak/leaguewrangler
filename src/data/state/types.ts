import { Session } from './session/types';

export * from './session/types';

export interface ApplicationState {
  readonly session: Session;
}
