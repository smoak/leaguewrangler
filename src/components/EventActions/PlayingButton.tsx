import React, { FC } from 'react';
import PlayingIcon from '@material-ui/icons/CheckCircleOutline';
import RsvpedPlayingIcon from '@material-ui/icons/CheckCircle';
import { IconButton } from '@material-ui/core';

interface PlayingButtonProps {
  readonly isPlaying: boolean;
  readonly className: string;
}

const PlayingButton: FC<PlayingButtonProps> = ({ isPlaying, className }) => {
  const icon = isPlaying ? <RsvpedPlayingIcon className={className} /> : <PlayingIcon className={className} />;

  return <IconButton aria-label="Playing">{icon}</IconButton>;
};

export default PlayingButton;
