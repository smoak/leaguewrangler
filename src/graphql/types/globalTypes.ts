export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: 'Query';
  /** Gets the current user */
  currentUser: User;
  /** Get a single event */
  event?: Maybe<Event>;
  /** Get events for a team */
  events: Array<Maybe<Event>>;
  /** Get teams for a user */
  teams: Array<Maybe<Team>>;
};


export type QueryEventArgs = {
  eventId: Scalars['Int'];
  teamId: Scalars['Int'];
};


export type QueryEventsArgs = {
  teamId: Scalars['Int'];
};


export type QueryTeamsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** A user */
export type User = {
  __typename?: 'User';
  displayName: Scalars['String'];
  emailAddress1: Scalars['String'];
  events: Array<Maybe<Event>>;
  firstName: Scalars['String'];
  fullName: Scalars['String'];
  lastName: Scalars['String'];
  profilePhotoFullUrl: Scalars['String'];
  profilePhotoSmallUrl: Scalars['String'];
  profilePhotoThumbnailUrl: Scalars['String'];
  userId: Scalars['Int'];
};


/** A user */
export type UserEventsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** An event */
export type Event = {
  __typename?: 'Event';
  endTimestamp?: Maybe<Scalars['Int']>;
  eventId: Scalars['Int'];
  eventType: Scalars['String'];
  location?: Maybe<Location>;
  seasonId: Scalars['Int'];
  seasonName?: Maybe<Scalars['String']>;
  startTimestamp?: Maybe<Scalars['Int']>;
  status: Scalars['String'];
  team: Team;
  title: Scalars['String'];
  viewerRsvpStatus: RsvpStatus;
};

/** A location */
export type Location = {
  __typename?: 'Location';
  address?: Maybe<Scalars['String']>;
  googleMapsDirectionsUrl?: Maybe<Scalars['String']>;
  googleMapsUrl?: Maybe<Scalars['String']>;
  isPublic: Scalars['Boolean'];
  locationId: Scalars['Int'];
  name: Scalars['String'];
};

/** A team */
export type Team = {
  __typename?: 'Team';
  name: Scalars['String'];
  photoFullUrl?: Maybe<Scalars['String']>;
  photoSmallUrl?: Maybe<Scalars['String']>;
  photoThumbnailUrl?: Maybe<Scalars['String']>;
  shortName: Scalars['String'];
  teamId: Scalars['Int'];
};

/** The RSVP status of a user */
export enum RsvpStatus {
  /** Indicates the user is available to attend */
  AVAILABLE = 'AVAILABLE',
  /** Indicates the user might attend */
  MAYBE = 'MAYBE',
  /** Indicates the user will not attend */
  NO = 'NO',
  /** Indicates the user has not RSVPed */
  NONE = 'NONE',
  /** Indicates the user will attend */
  YES = 'YES'
}

export type Mutation = {
  __typename?: 'Mutation';
  /** Create a user token */
  createUserToken?: Maybe<UserToken>;
  /** RSVP to an event */
  saveRsvp: Event;
};


export type MutationCreateUserTokenArgs = {
  password: Scalars['String'];
  username: Scalars['String'];
};


export type MutationSaveRsvpArgs = {
  comments?: InputMaybe<Scalars['String']>;
  eventId: Scalars['Int'];
  rsvpStatus: RsvpStatus;
  teamId: Scalars['Int'];
};

/** A user token */
export type UserToken = {
  __typename?: 'UserToken';
  /** A unique token for the user matched. Tokens are 36-character, lower-cased GUIDs. */
  token: Scalars['String'];
};
