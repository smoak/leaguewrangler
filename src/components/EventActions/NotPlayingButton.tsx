import { IconButton } from '@material-ui/core';
import NotPlayingIcon from '@material-ui/icons/HighlightOff';
import RsvpedNotPlayingIcon from '@material-ui/icons/HighlightOffTwoTone';
import React, { FC, useCallback } from 'react';

import { RsvpStatus } from '../../graphql/types/globalTypes';

interface NotPlayingButtonProps {
  readonly isNotPlaying: boolean;
  readonly className: string;
  onRsvp(rsvpStatus: RsvpStatus): void;
}

const NotPlayingButton: FC<NotPlayingButtonProps> = ({ className, isNotPlaying, onRsvp }) => {
  const rsvpCallback = useCallback(() => {
    onRsvp(RsvpStatus.NO);
  }, [onRsvp]);

  const icon = isNotPlaying ? (
    <RsvpedNotPlayingIcon className={className} color="primary" />
  ) : (
    <NotPlayingIcon className={className} />
  );

  return (
    <IconButton aria-label="Not Playing" onClick={rsvpCallback}>
      {icon}
    </IconButton>
  );
};

export default NotPlayingButton;
