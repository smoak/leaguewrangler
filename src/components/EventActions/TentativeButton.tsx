import { IconButton } from '@mui/material';
import TentativeIcon from '@mui/icons-material/HelpOutlineOutlined';
import RsvpedTentativeIcon from '@mui/icons-material/HelpTwoTone';
import { FC, useCallback } from 'react';

import { RsvpStatus } from '../../graphql/types/globalTypes';

interface TentativeButtonProps {
  readonly isTentative: boolean;
  readonly className: string;
  onRsvp(rsvpStatus: RsvpStatus): void;
}

const TentativeButton: FC<TentativeButtonProps> = ({ onRsvp, className, isTentative }) => {
  const rsvpCallback = useCallback(() => {
    onRsvp(RsvpStatus.MAYBE);
  }, [onRsvp]);

  const icon = isTentative ? (
    <RsvpedTentativeIcon className={className} color="primary" />
  ) : (
    <TentativeIcon className={className} />
  );
  return (
    <IconButton aria-label="Tentative" onClick={rsvpCallback} size="large">
      {icon}
    </IconButton>
  );
};

export default TentativeButton;
