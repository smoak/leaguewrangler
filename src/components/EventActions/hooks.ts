import { useCallback } from 'react';
import { useMutation } from 'react-apollo';

import saveEventRsvp from '../../graphql/mutations/saveRsvp';
import { SaveEventRsvp, SaveEventRsvpVariables } from '../../graphql/types/SaveEventRsvp';
import { RsvpStatus } from '../../graphql/types/globalTypes';

type SaveRsvpCallback = (rsvpStatus: RsvpStatus) => void;

interface SaveRsvpMutation {
  readonly loading: boolean;
  readonly saveRsvpCallback: SaveRsvpCallback;
}

interface UseSaveRsvpMutationOptions {
  readonly eventId: number;
  readonly teamId: number;
}

type UseSaveRsvpMutation = (options: UseSaveRsvpMutationOptions) => SaveRsvpMutation;
export const useSaveRsvpMutation: UseSaveRsvpMutation = ({ eventId, teamId }) => {
  const [saveRsvp, { loading }] = useMutation<SaveEventRsvp, SaveEventRsvpVariables>(saveEventRsvp);
  const saveRsvpCallback = useCallback(
    (rsvpStatus) => {
      saveRsvp({ variables: { eventId, teamId, rsvpStatus } });
    },
    [eventId, teamId, saveRsvp]
  );

  return {
    loading,
    saveRsvpCallback,
  };
};
