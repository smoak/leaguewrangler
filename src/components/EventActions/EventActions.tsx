import { CircularProgress, WithStyles, createStyles, withStyles } from '@material-ui/core';
import { FC } from 'react';

import { RsvpStatus } from '../../graphql/types/globalTypes';

import NotPlayingButton from './NotPlayingButton';
import PlayingButton from './PlayingButton';
import TentativeButton from './TentativeButton';
import { useSaveRsvpMutation } from './hooks';

interface OwnProps {
  readonly eventId: number;
  readonly rsvpStatus: RsvpStatus;
  readonly teamId: number;
}

const styles = createStyles({
  icon: {
    fontSize: 30,
  },
});

type EventActionsProps = WithStyles<typeof styles> & OwnProps;

export const EventActions: FC<EventActionsProps> = ({ classes, eventId, teamId, rsvpStatus }) => {
  const { loading, saveRsvpCallback } = useSaveRsvpMutation({ teamId, eventId });

  if (loading) {
    return <CircularProgress className={classes.icon} color="secondary" />;
  }

  return (
    <>
      <PlayingButton className={classes.icon} isPlaying={rsvpStatus === RsvpStatus.YES} onRsvp={saveRsvpCallback} />
      <TentativeButton
        className={classes.icon}
        isTentative={rsvpStatus === RsvpStatus.MAYBE}
        onRsvp={saveRsvpCallback}
      />
      <NotPlayingButton
        className={classes.icon}
        isNotPlaying={rsvpStatus === RsvpStatus.NO}
        onRsvp={saveRsvpCallback}
      />
    </>
  );
};

export default withStyles(styles)(EventActions);
