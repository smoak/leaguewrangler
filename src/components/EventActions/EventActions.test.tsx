import { ShallowWrapper, shallow } from 'enzyme';
import React from 'react';

import EventActions from './EventActions';
import { RsvpStatus } from 'graphql/types/globalTypes';

describe('EventActions', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<EventActions rsvpStatus={RsvpStatus.NONE} />);
  });

  it('renders as expected', () => {
    expect(component).toMatchSnapshot();
  });
});
