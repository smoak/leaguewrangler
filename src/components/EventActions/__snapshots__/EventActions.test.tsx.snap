// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventActions renders as expected 1`] = `
<EventActions
  classes={
    Object {
      "icon": "EventActions-icon-1",
    }
  }
  rsvpStatus="NONE"
/>
`;
